sklearn/discriminant_analysis.py

```diff
@@ -623,11 +623,16 @@ class QuadraticDiscriminantAnalysis(BaseEstimator, ClassifierMixin):
         Discriminant Analysis
     """
 
-    def __init__(self, priors=None, reg_param=0., store_covariances=False,
-                 tol=1.0e-4):
+    def __init__(self, priors=None, reg_param=0., store_covariance=False,
+                 tol=1.0e-4, store_covariances=False):
+        if store_covariances is not None:
```

Comment:
I'm not talking about the parameter passed into fit. I'm talking about the parameter passed in ``__init__``. That should be validated in ``fit`` not in ``__init__``.